### PR TITLE
Support remote updates to Target property token

### PR DESCRIPTION
**Pull request by swarna04**
## Description

Support remote property token updates with attach data rule:
The property token "at_property" can be provided in the JSON data to be attached to Target events (prefetch, execute, locations displayed or location clicked). Also, added the logic to remove any property tokens passed in mbox parameters (supported with Target iOS 2.x SDK).

The order or precedence will be as follows:
Property token configured in the extension card > Property token passed in Event Data

Note: Do not configure the property token in Target extension card on Data Collection UI if you wish to update it with attach data rule.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sbenedicadb** [comment]
> The order or precedence will be as follows:
Property token configured in the extension card > Property token passed in Event Data

Should we change this so the event data value has precedence over the configured value?  Seems like it's being used as an override, which in my mind means it should have higher priority.  WDYT?

**Comment by swarna04** [comment]
> > The order or precedence will be as follows:
> > Property token configured in the extension card > Property token passed in Event Data
> 
> Should we change this so the event data value has precedence over the configured value? Seems like it's being used as an override, which in my mind means it should have higher priority. WDYT?

I thought about this. And this is how we support it with 2.x to encourage customers to use the Launch card for property token configuration. Changing this behavior now might lead to more confusion. Moreover, we verified that LBG is not using the Target Launch configuration for property token. So they can just update it remotely for any Target request that they want. In case they do not update it for a request, no token will be attached (assume default workspace).

**Review by sbenedicadb — APPROVED** [review]
looks good - very minor stuff in comments

File: AEPTarget/Sources/Event+Target.swift [hunk 1]
```diff
@@ -24,6 +24,11 @@ extension Event {
         return TargetParameters.from(dictionary: data?[TargetConstants.EventDataKeys.TARGET_PARAMETERS] as? [String: Any])
     }
 
+    /// Reads the Target `at_property` from the event data
+    var propertyToken: String {
+        return data?[TargetConstants.EventDataKeys.AT_PROPERTY] as? String ?? ""
```
[the remark this answers is not in the document]
Comment (swarna04) [reply AEPTarget/Sources/Event+Target.swift:29]: Yup, thanks for the reminder!

**Pull request merged**